django/db/models/fields/json.py

```diff
@@ -75,9 +75,17 @@ def deconstruct(self):
     def from_db_value(self, value, expression, connection):
         if value is None:
             return value
+        if isinstance(expression, KeyTransform):
+            if connection.features.json_extract_returns_sql_scalar:
+                return value
```

Comment:
Won't this be an issue if `value` isn't a scalar? Or maybe the feature is misnamed?

Reply:
You're right, I missed that one. Thanks!